Sources/SwiftInspectorCore/Tests/InitializerAnalyzerSpec.swift

```diff
@@ -0,0 +1,135 @@
+// Created by Francisco Diaz on 3/27/20.
+//
+// Copyright (c) 2020 Francisco Diaz
+//
+// Distributed under the MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Nimble
+import Quick
+import Foundation
+
+@testable import SwiftInspectorCore
+
+final class InitializerAnalyzerSpec: QuickSpec {
+  override func spec() {
+    var fileURL: URL!
+    var sut = InitializerAnalyzer(name: "FakeType")
+
+    beforeEach {
+      sut = InitializerAnalyzer(name: "FakeType")
+    }
+
+    afterEach {
+      guard let fileURL = fileURL else {
+        return
+      }
+      try? Temporary.removeItem(at: fileURL)
+    }
+
+    describe("analyze(fileURL:)") {
+      context("when there is no initializer") {
+        beforeEach {
+          let content = """
+                        public final class FakeType {}
+                        """
+          fileURL = try? Temporary.makeFile(content: content, name: "ABC")
+        }
+
+        it("returns an empty array") {
+          let result = try? sut.analyze(fileURL: fileURL)
+
+          expect(result).to(beEmpty())
+        }
+      }
+
+      context("when there is an initializer") {
+        beforeEach {
+          let content = """
+          public final class FakeType {
+            convenience init(someString: String, someInt: Int) {}
```

Comment:
Can we add a test for an initializer that spans multiple lines?

Reply:
I don't think we need to. This relies on SwiftSyntax doing that for us, so we could add tests like that but IMO we're testing there `SwiftSyntax` ore than we're testing `SwiftInspector` which IMO we shouldn't be testing.

Reply:
I did think about this when I added my comment. I'm open to that idea. I guess this is where we gain a lot vs. using regexes 💪